tools/doc-generator/doc-generator.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bufio"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"sort"
+	"strings"
+
+	fake "kubevirt.io/kubevirt/pkg/monitoring/vms/prometheus"
+	promvm "kubevirt.io/kubevirt/pkg/monitoring/vms/prometheus" // import for prometheus metrics
+	_ "kubevirt.io/kubevirt/pkg/virt-controller/watch"
+)
+
+func main() {
+	req, err := http.NewRequest(http.MethodGet, "/metrics", nil)
+	checkError(err)
+
+	recorder := httptest.NewRecorder()
+
+	handler := promvm.Handler(1)
+
+	fake.RegisterFakeCollector()
+
+	handler.ServeHTTP(recorder, req)
+
+	if status := recorder.Code; status == http.StatusOK {
+		metrics := parseVirtMetrics(recorder.Body, []string{"kubevirt_", "_virt_controller"})
```

Comment:
I'd either make the string array constant or move it into the function so the reader can directly see what is checked for.